# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/tabs/src/sync/engine.rs
# Commit: 2147c5c81edf
# Full Hash: 2147c5c81edf9cd56c19e7e92dfebfcd77329978
# Author: Ryan VanderMeulen <ryanvm@gmail.com>
# Date: 2023-03-01 17:52 -0500
# Description:
#   Backed out changesets b05343a5b533 and 57b2cda7e0db (bug 1818349) for causing bug 1819723.
# ==============================================================================

diff -r d59b76766f0d -r 2147c5c81edf third_party/rust/tabs/src/sync/engine.rs
--- a/third_party/rust/tabs/src/sync/engine.rs	Wed Mar 01 05:53:30 2023 +0000
+++ b/third_party/rust/tabs/src/sync/engine.rs	Wed Mar 01 17:52:01 2023 -0500
@@ -311,7 +311,11 @@
             .unwrap()
             .apply_incoming(inbound.changes, telem)?;
 
-        Ok(OutgoingChangeset::new("tabs".into(), outgoing_records))
+        Ok(OutgoingChangeset::new_with_changes(
+            "tabs",
+            inbound.timestamp,
+            outgoing_records,
+        ))
     }
 
     fn sync_finished(
@@ -339,9 +343,7 @@
         Ok(if since == server_timestamp {
             vec![]
         } else {
-            vec![CollectionRequest::new("tabs".into())
-                .full()
-                .newer_than(since)]
+            vec![CollectionRequest::new("tabs").full().newer_than(since)]
         })
     }
 